api/unstable/ktls.h

```diff
@@ -0,0 +1,111 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file ktls.h
+ *
+ * The following APIs enable applications to use kernel TLS (kTLS), meaning that
+ * enrypting and decrypting TLS records is handled by the kernel rather than by
+ * the s2n-tls library.
+ *
+ * The kTLS APIs are currently considered unstable. kTLS is a relatively new
+ * feature with limited and volatile support from different kernels and hardware.
+ *
+ * Currently, s2n-tls supports ktls for only very limited scenarios:
+ * - You must be using Linux. We have not tested with other kernels.
+ * - Your kernel must support kTLS. For Linux, versions >4.13 should support kTLS.
+ * - You must negotiate TLS1.2. TLS1.3 support is blocked on kernel support for
+ *   TLS KeyUpdate messages.
+ * - You must negotiate AES128-GCM. Other ciphers will be supported soon.
```

Comment:
Thought we weren't making future promises ;) Also should we mention how users can negotiate that particular ciphersuite?

Reply:
I can replace "soon" with eventually then xD We kind of have to eventually support other ciphers, otherwise it'd be a pretty weird feature.